From 917ba981cd1e3901a77decc80e3dcce89682a42a Mon Sep 17 00:00:00 2001
From: Erico Nunes <ernunes@redhat.com>
Date: Fri, 3 Apr 2020 17:44:41 +0200
Subject: [PATCH] fwts: fix access_ok for the RHEL8 kernel

The RHEL8 4.18 kernel backported a fix from the 5.0 kernel which changes
the number of arguments for the access_ok macro.
fwts handles this for the upstream kernel by checking the kernel version
as 5.0 or bigger, but this fails for RHEL8 since it runs a 4.18 patched
kernel.
To fix the issue for RHEL8, use the modified macro for the RHEL8 kernel
kernel as well.

Signed-off-by: Erico Nunes <ernunes@redhat.com>
---
 efi_runtime/efi_runtime.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/efi_runtime/efi_runtime.c b/efi_runtime/efi_runtime.c
index 5a308c31..ae916f6c 100644
--- a/efi_runtime/efi_runtime.c
+++ b/efi_runtime/efi_runtime.c
@@ -41,7 +41,9 @@ MODULE_LICENSE("GPL");
 #define EFI_RUNTIME_ENABLED	efi_enabled
 #endif
 
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 0, 0)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 0, 0) || \
+    (defined(RHEL_MAJOR) && defined(RHEL_MINOR) && \
+     RHEL_MAJOR == 8 && RHEL_MINOR >= 1)
 #define ACCESS_OK(type, addr, size)	access_ok(addr, size)
 #else
 #define ACCESS_OK(type, addr, size)	access_ok(type, addr, size)
-- 
2.25.1

